File: SDKs/amazonaws.com/opsworks/2013-02-18/java/lib/src/main/java/openapisdk/models/shared/EbsBlockDevice.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * EbsBlockDevice
 * Describes an Amazon EBS volume. This data type maps directly to the Amazon EC2 <a href="https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_EbsBlockDevice.html">EbsBlockDevice</a> data type.
**/
public class EbsBlockDevice {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("DeleteOnTermination")
    public Boolean deleteOnTermination;
    public EbsBlockDevice withDeleteOnTermination(Boolean deleteOnTermination) {
        this.deleteOnTermination = deleteOnTermination;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Iops")
    public Long iops;
    public EbsBlockDevice withIops(Long iops) {
        this.iops = iops;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("SnapshotId")
    public String snapshotId;
    public EbsBlockDevice withSnapshotId(String snapshotId) {
        this.snapshotId = snapshotId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("VolumeSize")
    public Long volumeSize;
    public EbsBlockDevice withVolumeSize(Long volumeSize) {
        this.volumeSize = volumeSize;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("VolumeType")
    public VolumeTypeEnum volumeType;
    public EbsBlockDevice withVolumeType(VolumeTypeEnum volumeType) {
        this.volumeType = volumeType;
        return this;
    }
}

File: SDKs/amazonaws.com/opsworks/2013-02-18/python/src/sdk/models/shared/ebsblockdevice.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import volumetype_enum as shared_volumetype_enum


@dataclass_json
@dataclasses.dataclass
class EbsBlockDevice:
    r"""EbsBlockDevice
    Describes an Amazon EBS volume. This data type maps directly to the Amazon EC2 <a href=\"https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_EbsBlockDevice.html\">EbsBlockDevice</a> data type.
    """
    
    delete_on_termination: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DeleteOnTermination') }})
    iops: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Iops') }})
    snapshot_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('SnapshotId') }})
    volume_size: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('VolumeSize') }})
    volume_type: Optional[shared_volumetype_enum.VolumeTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('VolumeType') }})
    
